lux-lua/src/config.rs

```diff
@@ -0,0 +1,97 @@
+use lux_lib::config::ConfigBuilder;
+use mlua::{ExternalResult, Lua, Table};
+
+pub fn config(lua: &Lua) -> mlua::Result<Table> {
+    let table = lua.create_table()?;
+
+    table.set(
+        "default",
+        lua.create_function(|_, ()| ConfigBuilder::default().build().into_lua_err())?,
+    )?;
+
+    table.set(
+        "builder",
+        lua.create_function(|_, ()| Ok(ConfigBuilder::default()))?,
+    )?;
+
+    Ok(table)
+}
+
+#[cfg(test)]
+mod tests {
+    use mlua::prelude::*;
+
+    #[test]
+    fn lua_api_test_config() {
+        let lua = Lua::new();
+
+        lua.globals().set("lux", crate::lux(&lua).unwrap()).unwrap();
+
+        lua.load(
+            r#"
+            local config = lux.config
+            local default = config.default()
+            assert(default, "default config should not be nil")
+            "#,
+        )
+        .exec()
+        .unwrap();
+    }
+
+    #[test]
+    fn lua_api_test_config_builder() {
+        let lua = Lua::new();
+        let tree = assert_fs::TempDir::new().unwrap();
+        let cache = assert_fs::TempDir::new().unwrap();
+        let data = assert_fs::TempDir::new().unwrap();
+
+        lua.globals().set("lux", crate::lux(&lua).unwrap()).unwrap();
+        lua.globals().set("tree", tree.path()).unwrap();
+        lua.globals().set("cache", cache.path()).unwrap();
+        lua.globals().set("data", data.path()).unwrap();
+
+        lua.load(
+            r#"
+            local config = lux.config
+            local default = config.builder()
```

Comment:
quibble: I suppose this isn't actually the default config?

```suggestion
            local config = lux.config.builder()
```